src/articles/OwnArticles.js

```diff
@@ -12,14 +14,63 @@ import * as s from "../components/TopMessage.sc";
 export default function OwnArticles({ api }) {
   const [articleList, setArticleList] = useState(null);
   const [originalList, setOriginalList] = useState(null);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [isWaitingForNewArticles, setIsWaitingForNewArticles] = useState(false);
+  const [noMoreArticlesToShow, setNoMoreArticlesToShow] = useState(false);
+
+  const articleListRef = useShadowRef(articleList);
+  const currentPageRef = useShadowRef(currentPage);
+  const noMoreArticlesToShowRef = useShadowRef(noMoreArticlesToShow);
+  const isWaitingForNewArticlesRef = useShadowRef(isWaitingForNewArticles);
+
+  function insertNewArticlesIntoArticleList(
+    fetchedArticles,
+    newCurrentPage,
+    newArticles,
+  ) {
+    if (fetchedArticles.length === 0) {
+      setNoMoreArticlesToShow(true);
+    }
+    newArticles = newArticles.concat(fetchedArticles);
+    setArticleList(newArticles);
+    setCurrentPage(newCurrentPage);
+    setIsWaitingForNewArticles(false);
+  }
+
+  function handleScroll() {
```

Comment:
there's a bit of duplication now between `OwnArticles` and `FindArticles`. I think we can live with it, but at the same time, if there was an easy way to extract a `useArticlePagination` hook from the two it might be good to do. But if it's too complicated, then it's fine.

Reply:
Indeed, I had thought about this - but I was a bit unsure about how to do it in a way that would allow for different behaviours depending on which component we are using the pagination from. 

I have just pushed what I ended up coding - let me know what you think